src/components/TokenBottomSheet.tsx

```diff
@@ -160,23 +166,52 @@ function TokenBottomSheet<T extends TokenBalance>({
 
   const { t } = useTranslation()
 
-  const handleToggleFilterChip = (toggledChip: FilterChip<TokenBalance>) => {
-    ValoraAnalytics.track(TokenBottomSheetEvents.toggle_tokens_filter, {
-      filterId: toggledChip.id,
-      isRemoving: filters.find((chip) => chip.id === toggledChip.id)?.isSelected ?? false,
-      isPreSelected: filterChips.find((chip) => chip.id === toggledChip.id)?.isSelected ?? false,
-    })
+  const networkChipRef = useRef<BottomSheetRefType>(null)
+  const networkChip = useMemo(
+    () => filters.find((chip): chip is NetworkFilterChip<TokenBalance> => isNetworkChip(chip)),
+    [filters]
+  )
 
+  // These function params mimic the params of the setSelectedNetworkIds function in
+  // const [selectedNetworkIds, setSelectedNetworkIds] = useState<NetworkId[]>([])
+  // This custom function is used to keep the same shared state between the network filter and the other filters
+  // which made the rest of the code more readable and maintainable
+  const setSelectedNetworkIds = (arg: NetworkId[] | ((networkIds: NetworkId[]) => NetworkId[])) => {
     setFilters((prev) => {
       return prev.map((chip) => {
-        if (chip.id === toggledChip.id) {
-          return { ...chip, isSelected: !chip.isSelected }
+        if (isNetworkChip(chip)) {
+          const selectedNetworkIds = typeof arg === 'function' ? arg(chip.selectedNetworkIds) : arg
+          return {
+            ...chip,
+            selectedNetworkIds,
+            isSelected: selectedNetworkIds.length !== chip.allNetworkIds.length,
+          }
         }
         return chip
       })
     })
   }
 
+  const handleToggleFilterChip = (toggledChip: FilterChip<TokenBalance>) => {
+    if (isNetworkChip(toggledChip)) {
+      networkChipRef.current?.snapToIndex(0)
```

Comment:
this will never be invoked on close right? like the boolean filters?

Reply:
It's only called when the chip is pressed.